g2p/transducer/__init__.py

```diff
@@ -723,20 +823,20 @@ def check(
                 display_warnings=display_warnings,
                 original_input=tg.input_string,
             )
-        else:
-            result = True
-            for i, transducer in enumerate(self._transducers):
-                if not transducer.check(
-                    tg._tiers[i],
-                    display_warnings=display_warnings,
-                    original_input=tg.input_string,
-                ):
-                    # Don't short circuit if warnings are required
-                    if display_warnings:
-                        result = False
-                    else:
-                        return False
-            return result
+        # if not shallow, go deeper
+        result = True
+        for i, transducer in enumerate(self._transducers):
+            if not transducer.check(
+                tg._tiers[i],
+                display_warnings=display_warnings,
+                original_input=tg.input_string,
+            ):
+                # Don't short circuit if warnings are required
+                if display_warnings:
+                    result = False
+                else:
+                    return False
+        return result
```

Comment:
I'm sorry, but I strongly disagree with this change. The pylint warning `no-else-return` apparently comes from the Mozilla coding style guide (at least according to https://stackoverflow.com/a/55561880) but I think it should be disabled. I have disabled it in `g2p/.pylintrc` with `disable=no-else-return`, but maybe VSCode doesn't use that config file. Would you mind configuring your VSCode to disable it globally?

This blog post sums up my position on this very well: https://dmerej.info/blog/post/else-after-return-yea-or-nay/ (found at https://stackoverflow.com/a/57539298):
 - When you have a parallel structure where one block is "if a" and the other block if conceptually "if not a", and both blocks return, the `else` displays that visually to the programmer. While the else might be redundant to the interpreter, cognitively it is not redundant at all: it communicates the intent of the code to the next programmer reading the code.
 - When you have a guard block, like "if bad situation: return" then it makes sense not to use an else, because the rest of the code is not actually in a parallel structure, it's just unable to proceed when the guard fails. So an else here would distract the programmer reading the code rather than add clarity.

In this case, I believe you actually agreed with me, because you felt the need to add the comment "if not shallow, go deeper", which tells me that you thought following pylint's suggestion made the code less clear.